### PR TITLE
MLB-1797 Added titles for payments and collection plan on CheckoutScreen

### PR DESCRIPTION
**Pull request by jlplks**
# 📲 What

Added titles for payments and collection plan component on CheckoutScreen when plot enabled

# 🤔 Why

When plot is enabled it should render Collection Plan component along with a title for Payments component

# 🛠 How

- Added prop validation for plotEnabled on CheckOutScreen, render title and collection plan component if plorEnabled is true, fix some styles of collection plan component

# 👀 See

Screenshot 2024-11-12 at 5 05 30 p m
Screenshot 2024-11-12 at 5 05 42 p m

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/CheckoutScreen.kt [hunk 1]
```diff
@@ -147,7 +192,8 @@ fun CheckoutScreen(
     newPaymentMethodClicked: () -> Unit,
     onDisclaimerItemClicked: (disclaimerItem: DisclaimerItems) -> Unit,
     onAccountabilityLinkClicked: () -> Unit,
-    onChangedPaymentMethod: (StoredCard?) -> Unit = {}
+    onChangedPaymentMethod: (StoredCard?) -> Unit = {},
+    plotEnabled: Boolean = true
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/CheckoutScreen.kt:196]: Is plot supposed to be enabled by default here?

Comment (Arkariang) [reply app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/CheckoutScreen.kt:196]: By default should be false, on the VM if you like @jlplks you can already send for this field the value of the FF, which is should be currently off.

Comment (jlplks) [reply app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/CheckoutScreen.kt]: Comments addressed

File: app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/ProjectPledgeButtonAndFragmentContainer.kt [hunk 2]
```diff
@@ -291,7 +292,9 @@ fun ProjectPledgeButtonAndFragmentContainer(
                                         newPaymentMethodClicked = onAddPaymentMethodClicked,
                                         isLoading = isLoading,
                                         onDisclaimerItemClicked = onDisclaimerItemClicked,
-                                        onAccountabilityLinkClicked = onAccountabilityLinkClicked
+                                        onAccountabilityLinkClicked = onAccountabilityLinkClicked,
+                                        isPlotEnabled = environment?.featureFlagClient()
+                                            ?.getBoolean(FlagKey.ANDROID_PLEDGE_OVER_TIME) ?: false,
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/ProjectPledgeButtonAndFragmentContainer.kt:297]: should check for pledge reason here as well

File: app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/CheckoutScreen.kt [hunk 3]
```diff
@@ -215,7 +261,9 @@ fun CheckoutScreen(
                                     .fillMaxWidth(),
                                 onClickAction = { onPledgeCtaClicked(selectedOption.value) },
                                 isEnabled = project.acceptedCardType(selectedOption.value?.type()) || selectedOption.value?.isFromPaymentSheet() ?: false,
-                                text = if (pledgeReason == PledgeReason.PLEDGE || pledgeReason == PledgeReason.LATE_PLEDGE) stringResource(id = R.string.Pledge) + " $totalAmountString" else stringResource(
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/CheckoutScreen.kt:218]: @jlplks is plot gonna be enabled for late pledges as well?

Comment (Arkariang) [reply app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/CheckoutScreen.kt:218]: Confirmed, is only gonna be available for new pledges, not for late pledges!

File: app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/ProjectPledgeButtonAndFragmentContainer.kt (hunk 2 above)
Comment (Arkariang) [reply app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/ProjectPledgeButtonAndFragmentContainer.kt:297]: Actually for late pledges flow should always be false.  Is gonna be an available payment option only for new pledges while in campaign

Comment (jlplks) [reply app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/ProjectPledgeButtonAndFragmentContainer.kt:297]: changin it rn

**Review by leighdouglas — APPROVED** [review]

**Pull request merged**